cmd/skopeo/sync.go

```diff
@@ -439,60 +440,60 @@ func filterSourceReferences(sys *types.SystemContext, registryName string, colle
 			continue
 		}
 
-		sourceReferences = filter(logger, sourceReferences)
-		if len(sourceReferences) == 0 {
+		var filteredSourceReferences []types.ImageReference
+		for _, ref := range sourceReferences {
+			if filter(logger, ref) {
+				filteredSourceReferences = append(filteredSourceReferences, ref)
```

Comment:
`break` here, so that multiple filters don’t cause the same image to be copied twice.

Reply:
adding the `break` here made the program stop after finding the first tag that match regex or semver

Reply:
I think you meant `continue` instead of `break`? it worked with `continue`

Reply:
I think I was just wrong — for some reason I thought this is iterating over filters, not over tags.

`continue` does nothing at the ~end of a loop.

Reply:
@mtrmac I think the code is now ready for your review